### PR TITLE
Extend CI pipelines to build against Herbert branch or tag

### LINKED ISSUE #332: Extend CI branch pipeline to build against Herbert branch or tag
The branch pipelines build against the last successful `master` build of Herbert.

To support release- and feature-branch based builds this should be modified to build against a specified branch name or tag.
The build-against-Herbert-source piece of the PR pipeline can be reused.

Output
- [ ] new Herbert branch/tag parameter added to pipeline [default `master`]
- [ ] pull Herbert sources for that branch/tag
- [ ] build Herbert from source
- [ ] resume existing pipeline building and testing Horace

### PR DESCRIPTION
**Pull request by hsautessella**
This PR adds a `HERBERT_BRANCH` parameter to our Jenkins pipelines. When triggering builds we now have the option to select a given Herbert branch or tag to clone from GitHub. *Nightly builds will still use the last successful Herbert nightly package*.

This PR also moves the parameter definitions in the Horace pipelines into the Jenkinsfile. This better documents what our parameters are and keeps our pipeline in sync with one another.

Fixes #332

### CODE REVIEW

File: documentation/smg/08_setting_up_jenkins_ci.md [hunk 1]
```diff
@@ -1,43 +1,59 @@
...
-In order to build and run, Horace requires that Herbert also be present in the Jenkins
-workspace. So that Horace builds are run against an up-to-date Herbert,
+In order to build and run, Horace requires that Herbert also be present in the
+Jenkins workspace.
+So that Horace builds are run against an up-to-date Herbert,
 pull request and branch builds will checkout and build the latest Herbert master from GitHub.
-This means that Horace is run against Herbert source, instead of a Herbert package.
+This means that Horace is run against Herbert source,
+instead of a Herbert package.
+There also exists a parameter that allows one to build a particular branch or
```
Comment (nickbattam-tessella) [inline documentation/smg/08_setting_up_jenkins_ci.md:55]: which parameter? you may as well name it

**Comment by nickbattam-tessella** [comment]
New issue created to document the new credentials keys: https://github.com/pace-neutrons/Horace/issues/351

File: documentation/smg/08_setting_up_jenkins_ci.md [hunk 2]
```diff
@@ -87,28 +116,22 @@ and should be prefixed with `PR-` if the pipeline is building pull requests, e.g
 `PR-Scientific-Linux-7-2018b`.
 - Enter the GitHub project URL e.g. `http:/pace-neutrons.github.com/horace`
 (this creates a link to the GitHub from the pipeline).
-- Select the `This project is parameterised` option:
```
Comment (nickbattam-tessella) [inline documentation/smg/08_setting_up_jenkins_ci.md:90]: You don't mention `BRANCH-` prefix for branch builds here

**Comment by hsautessella** [comment]
Pipelines using the updated Jenkinsfile have been tested on pipelines such as `Branch-Windows-10-TESTER-2018b`. These pipelines have since been deleted.

**Review by nickbattam-tessella — APPROVED** [review]

**Pull request merged**